runtime/domain.c

```diff
@@ -1489,16 +1491,15 @@ void caml_interrupt_self(void) {
 
 void caml_reset_young_limit(caml_domain_state * dom_st)
 {
-  CAMLassert(dom_st == Caml_state);
   /* An interrupt might have been queued in the meanwhile; this
      achieves the proper synchronisation. */
   atomic_exchange(&dom_st->young_limit, (uintnat)dom_st->young_start);
-  if (caml_incoming_interrupts_queued()
+  dom_internal * d = &all_domains[dom_st->id];
```

Comment:
This changes seems to be paving the way to call `caml_reset_young_limit` at another domain, but I can't tell (neither in #11057) where that would happen. When would we need it? (Are you planning to handle cross-domain interrupts with this mechanism, by calling `caml_resete_young_limit` on the thread to interrupt?)

Reply:
"private communication" from @sadiqj.

I think this is a desired direction for the domains implementation, and the current style of the surrounding code.

Reply:
Are you sure this was me? I'm not sure I recognise that code.

Reply:
It was actually a quote from @ctk21 which you forwarded to me. It was about passing the domain state around rather than introducing calls to Caml_state (something about work stealing for STW sections, and also the cost of Caml_state on some platforms).